fafbot.py

```diff
@@ -27,332 +29,46 @@
-HITBOX_STREAMS = "https://www.hitbox.tv/api/media/live/list?filter=popular&game=811&hiddenOnly=false&limit=30&liveonly=true&media=true"
-
-class betmatch(object):
-    def __init__(self, uid, startTime, name, odds, mostProbableWinner):
-        try:
-            self.uid = uid
-            self.startTime = startTime
-            self.pot = 0
-            self.players = {}
-            self.betOn = {}
-            self.name = name
-            self.odds = odds
-            self.mostProbableWinner = mostProbableWinner
-        except:
-            pass
-            
-
-    def computeReward(self, amount):
-        try:
-            maxTime = (30.0 * 60.0)
-            minTime = (2 * 60)
-            if (time.time() - self.startTime) > minTime:
-                return amount*(1.0-((time.time() - self.startTime) / maxTime))
-            else:
-                return amount
-        except:
-            return 0
-
-    def addBeter(self, amount, uid, betUid):
-        try:
-            maxBet = self.computeReward(amount)
-            if not uid in self.players :
-                self.players[uid] = maxBet
-            if not betUid in self.betOn:
-                self.betOn[betUid] =  []
-            self.betOn[betUid].append(uid)
-            return maxBet
-        except:
-            return 0
-
-    def getBetWinners(self, uid):
-        try:
-            if uid in self.betOn :
-                return self.betOn[uid]
-            else:
-                return []
-        except:
-            return []
-
-    def getReward(self, uid):
-        try:
-            if uid in self.players:
-                return self.players[uid]
-            return 0
-        except:
-            return 0
-
-class bettingSystem(object):
-    def __init__(self):
-        self.matches = {}
-
-    def addMatch(self, match):
-        try:
-            if not match.uid in self.matches:
-                self.matches[match.uid] = match
-            return self.matches[match.uid]
-        except:
-            return None
-
-    def getMatches(self):
-        try:
-            return self.matches
-        except:
-            return None
-
-    def deleteMatch(self, uid):
-        try:
-            if uid in self.matches:
-                del self.matches[uid]
-        except:
-            pass
-
 
 class BotModeration(ircbot.SingleServerIRCBot):
     def __init__(self):
-        """
-        Constructeur qui pourrait prendre des parametres dans un "vrai" programme.
-        """
         # FIXME: hardcoded ip
-        ircbot.SingleServerIRCBot.__init__(self, [("37.58.123.2", 6667)],
-                                           "fafbot", "FAF bot")
+        ircbot.SingleServerIRCBot.__init__(
+            self,
+            [("37.58.123.2", 6667)],
+            "fafbot",
+            "FAF bot"
+        )
+
         self.nickpass = fafbot_config['nickpass']
         self.nickname = fafbot_config['nickname']
 
+        self.message_handler = MessageHandler(fafbot_config)
```

Comment:
It may seperate the concern of where to dispatch a message from the main bot class. But given that this is python, we can do that pretty elegantly in a single line:

`getattr(self, cmd_name)(...)`

The 'object graph needed to handle messages and delegate' has already been constructed by the interpreter when it parsed the program, so we are essentially taking something that the interpreter already does and reimplementing it. This could be fine if you need some fancy bells and whistles at dispatching, but I am not convinced of that being the case (Again, this is a straight forward simple irc bot).